### PR TITLE
[RPD-286] Add documentation on "stacks" and the new LLM stack

### PR DESCRIPTION
**Pull request by Christopher-Norman**
This PR adds a brief section into the docs about "stacks" and what is included in the LLM stack.

## Checklist

Please ensure you have done the following:

* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [x] I have updated the documentation if required.
* [ ] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [ ] Bug Fix (non-breaking change, fixing an issue)
* [ ] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Other (add details above)

### CODE REVIEW

**Review by JonoCX — CHANGES_REQUESTED** [review]
Great start 🌯 See comments

File: docs/resources.md
[the remark this answers is not in the document]
Comment (Christopher-Norman) [reply docs/resources.md:1]: Good shout, updated now

File: docs/resource-stacks.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+Matcha accommodates both of these requirements, and currently offers two infrastructure stacks which we'll discuss in more detail here and show how you can get started with either.
+
+> Note: These stacks must be set before provisioning any resources and cannot currently be change whilst a deployed.
+
+## Available stacks
+
+### DEFAULT
+
+The `DEFAULT` stack. This stack is ideal for generic machine learning training and deployments and a good starting point. It includes:
```
Comment (JonoCX) [inline docs/resource-stacks.md:13]: Perhaps we could point to the getting started page in this section, as this is the stack that's used there

File: docs/resource-stacks.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+The `LLM` stack: This includes everything found within the `DEFAULT` stack with the addition of a vector database - Chroma DB. This stack is modified for the training and deployment of Large Language Models (LLMs).
+
+   * [Azure Kubernetes Service](https://azure.microsoft.com/en-gb/products/kubernetes-service)
+   * [ZenML](https://www.zenml.io/home)
+   * [Seldon Core](https://www.seldon.io/solutions/open-source-projects/core) (deployment)
+   * [MLflow](https://mlflow.org/) (experiment tracking)
+   * [Data Version Control](https://dvc.org/) storage
+   * [Chroma DB](https://www.trychroma.com/) (vector database for document retrieval)
+
+## How to switch your stack
```
Comment (JonoCX) [inline docs/resource-stacks.md:32]: Point to the API documentation here too?

**Review by JonoCX — CHANGES_REQUESTED** [review]
Nice! 

It feels like the stack subsections are missing something, perhaps more detail a little detail about the two and their components. Plus I think we can link to other parts of the documentation here too, like the API and getting started.

File: docs/resource-stacks.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Resource Stacks 📚
+
+Machine Learning projects often vary in their size, from small-scale experimentation to large deployments, meaning that the infrastructure requirements also change and scale. For example, the infrastructure stack needed for deploying an LLM may require a GPU or vector database, which aren't usually needed in more general machine learning use-cases.
+
+Matcha accommodates both of these requirements, and currently offers two infrastructure stacks which we'll discuss in more detail here and show how you can get started with either.
+
+> Note: These stacks must be set before provisioning any resources and cannot currently be change whilst a deployed.
```
Comment (KirsoppJ) [inline docs/resource-stacks.md:7]: "whilst a deployed" needs rephrasing

File: docs/resource-stacks.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+## Available stacks
+
+### DEFAULT
+
+The `DEFAULT` stack. This stack is ideal for generic machine learning training and deployments and a good starting point. It includes:
+   * [Azure Kubernetes Service](https://azure.microsoft.com/en-gb/products/kubernetes-service)
+   * [ZenML](https://www.zenml.io/home)
+   * [Seldon Core](https://www.seldon.io/solutions/open-source-projects/core) (deployment)
+   * [MLflow](https://mlflow.org/) (experiment tracking)
+   * [Data Version Control](https://dvc.org/) storage
```
Comment (KirsoppJ) [inline docs/resource-stacks.md:18]: Don't think we need the link to dvc.org here. The data version control storage is just storage, nothing is installed there.

**Review by KirsoppJ — CHANGES_REQUESTED** [review]
Looking great, just a couple of niggles

**Review by JonoCX — APPROVED** [review]
LGTM

**Pull request merged**